pkg/operator/credentialsrequest/credentialsrequest_controller_test.go

```diff
@@ -1325,6 +1325,55 @@ func TestCredentialsRequestReconcile(t *testing.T) {
 			},
 			expectErr: true,
 		},
+		{
+			name: "root creds updated in passthrough mode",
+			existing: func() []runtime.Object {
+				objects := []runtime.Object{}
+				objects = append(objects, testOperatorConfig(operatorv1.CloudCredentialsModePassthrough))
+				objects = append(objects, createTestNamespace(testNamespace))
+				objects = append(objects, createTestNamespace(testSecretNamespace))
+
+				cr := testPassthroughCredentialsRequest(t)
+				cr.Generation = cr.Generation + 1                       // rev the generation to trigger the sync
+				cr.Status.LastSyncCloudCredsSecretResourceVersion = "1" // resource version of root creds before update
+				objects = append(objects, cr)
+
+				cloudCredsSecret := testPassthroughAWSCredsSecret("kube-system", "aws-creds", testRootAWSAccessKeyID, testRootAWSSecretAccessKey)
+				cloudCredsSecret.ResourceVersion = "2" // resource version of root creds after update
+				objects = append(objects, cloudCredsSecret)
+				objects = append(objects, testAWSCredsSecret("openshift-cloud-credential-operator", "cloud-credential-operator-iam-ro-creds", testReadAWSAccessKeyID, testReadAWSSecretAccessKey))
+				objects = append(objects, testAWSCredsSecret(testSecretNamespace, testSecretName, testAWSAccessKeyID, testAWSSecretAccessKey))
+				objects = append(objects, testClusterVersion())
+				objects = append(objects, testInfrastructure(testInfraName))
+
+				return objects
+			}(),
+			mockRootAWSClient: func(mockCtrl *gomock.Controller) *mockaws.MockClient {
+				mockAWSClient := mockaws.NewMockClient(mockCtrl)
+				return mockAWSClient
+			},
+			mockReadAWSClient: func(mockCtrl *gomock.Controller) *mockaws.MockClient {
+				mockAWSClient := mockaws.NewMockClient(mockCtrl)
+				mockGetUser(mockAWSClient)
+				return mockAWSClient
+			},
+			mockSecretAWSClient: func(mockCtrl *gomock.Controller) *mockaws.MockClient {
+				mockAWSClient := mockaws.NewMockClient(mockCtrl)
+				mockGetUser(mockAWSClient)
+				return mockAWSClient
+			},
+			validate: func(c client.Client, t *testing.T) {
+				targetSecret := getSecret(c)
+				if assert.NotNil(t, targetSecret) {
+					assert.Equal(t, testRootAWSAccessKeyID,
+						string(targetSecret.Data["aws_access_key_id"]))
+					assert.Equal(t, testRootAWSSecretAccessKey,
+						string(targetSecret.Data["aws_secret_access_key"]))
+				}
+				cr := getCR(c)
+				assert.True(t, cr.Status.Provisioned)
```

Comment:
add an assert here that the lastSyncCloudCredsSecretResource version was updated as expected.